File: 08-DesignPatterns-Java/Solution/State/src/States/NeutralState.java
package States;

public final class NeutralState extends PartnerStateBase {

    public NeutralState(Partner.Partner partner) {
        super(partner);
    }

    @Override
    public void talk() {
        print("Hello!");
    }

    @Override
    public void kiss() {
        print("Thanks! :-)");
        setState(new HappyState(_partner));
    }

    @Override
    public void insult() {
        print("Are you f***ing kidding me?!");
        setState(new AngryState(_partner));
    }
}

File: 09-DesignPatterns-CSharp/09-1-2-State-Solution/States/NeutralState.cs
namespace State.States
{
    internal sealed class NeutralState : PartnerStateBase
    {
        public NeutralState(Partner partner)
            : base(partner)
        {
        }

        public override void Talk()
        {
            Print("Hello!");
        }

        public override void Kiss()
        {
            Print("Thanks! :-)");
            SetState(new HappyState(Partner));
        }

        public override void Insult()
        {
            Print("Are you f***ing kidding me?!");
            SetState(new AngryState(Partner));
        }
    }
}
